test/fixtures/ancestors1/shadow.html

```diff
@@ -0,0 +1,53 @@
+<!--
+  Copyright (c) 2019, salesforce.com, inc.
+  All rights reserved.
+  SPDX-License-Identifier: BSD-3-Clause
+  For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+-->
+
+<body>
+  <script>
+    class OuterComponent extends HTMLElement {
+      constructor() {
+        super()
+        const shadowRoot = this.attachShadow({ mode: 'open' })
+        shadowRoot.innerHTML = `
+        <div>
+        </div>
+      `
+      }
+
+      connectedCallback() {
+        this.shadowRoot.querySelector('div').appendChild(new InnerComponent())
+      }
+    }
+
+    class InnerComponent extends HTMLElement {
+      constructor() {
+        super()
+        const shadowRoot = this.attachShadow({ mode: 'open' })
+        shadowRoot.innerHTML = `
+        <div class="container">
+      <div>
+        <div class="inner-component">
+          <label></label>
+          <select class="selectBox1"></select>
+        </div>
+      </div>
+      <div>
+        <div class="inner-component">
+          <label></label>
+          <select class="selectBox2"></select>
+        </div>
+      </div>
+    </div>
+            `;
```

Comment:
This is not really in the spirit of the original tests, since the classes marked `inner-component` do not have shadow roots immediately inside of them. It looks like you are effectively trying to do a light-dom test here. 🙂

Reply:
I'd say let's do one of the following:

1. Make this `shadow.html` test more like the other `shadow.html` tests – anything marked as a `*component` should be a component with a shadow root inside of it
2. Remove `shadow.html` and just run all the tests (kagekiri and `document.querySelector*`) against your `light.html` tests. For what you're trying to test, shadow DOM doesn't really matter anyway.